src/Library/demos/Wallpaper/main.js

```diff
@@ -0,0 +1,41 @@
+import Gtk from "gi://Gtk";
+import Gdk from "gi://Gdk";
+import Gio from "gi://Gio";
+import Xdp from "gi://Xdp";
+import XdpGtk from "gi://XdpGtk4";
+
+const portal = new Xdp.Portal();
+const parent = XdpGtk.parent_new_gtk(workbench.window);
+const button = workbench.builder.get_object("button");
+
+Gio._promisify(Gtk.FileDialog.prototype, "open", "open_finish");
+Gio._promisify(Xdp.Portal.prototype, "set_wallpaper", "set_wallpaper_finish");
+
+const filter = new Gtk.FileFilter({
+  name: "File",
+});
+
+const dialog = new Gtk.FileDialog({
```

Comment:
we only want to let the user select an image :)
an other option is to ship an image with Workbench and use that one 

We already have one, check the Overlay entry. I think it might make the entry clearer that any image can be set, not just something selected by the user.